core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -256,7 +256,14 @@ object SparkEnv extends Logging {
       if (rpcEnv.isInstanceOf[AkkaRpcEnv]) {
         rpcEnv.asInstanceOf[AkkaRpcEnv].actorSystem
       } else {
-        val actorSystemPort = if (port == 0) 0 else rpcEnv.address.port + 1
```

Comment:
It's simpler: `val actorSystemPort = if (port == 0 || rpcEnv.address == null) port else port + 1`